xbmc/music/MusicDatabase.cpp

```diff
@@ -3794,11 +3794,17 @@ bool CMusicDatabase::GetSongsFullByWhere(const std::string &baseDir, const Filte
     unsigned int time = XbmcThreads::SystemClockMillis();
     int total = -1;
 
-    std::string strSQL = "SELECT %s FROM songview ";
-    if (artistData) 
+    std::string strSQL = "SELECT songview.* FROM songview ";
+    if (artistData)
```

Comment:
Isn't better to move the whole strSQL initialization after you have strSQLExtra? So you don't create the string twice in some case. something like
```
std::string strSQL;
if (!artistdata)
  strSQL = "something1";
else if (!limited)
  strSQL = "something2";
else
  strSQL = "something3";
```
after you build the limit clause